src/Koffee/MainView.fs

```diff
@@ -227,8 +230,8 @@ module MainView =
             Bind.view(<@ window.SearchCaseSensitive.IsChecked @>).toModel(<@ model.SearchCaseSensitive @>, ((=) (Nullable true)), Nullable)
             Bind.view(<@ window.SearchRegex.IsChecked @>).toModel(<@ model.SearchRegex @>, ((=) (Nullable true)), Nullable)
             Bind.view(<@ window.SearchSubFolders.IsChecked @>).toModel(<@ model.SearchSubFolders @>, ((=) (Nullable true)), Nullable)
-            Bind.modelMulti(<@ model.InputMode, model.InputTextSelection, model.SelectedItem, model.PathFormat, model.Config.Bookmarks @>)
-                .toFunc(fun (inputMode, (selectStart, selectLen), selected, pathFormat, bookmarks) ->
+            Bind.modelMulti(<@ model.InputMode, model.InputTextSelection, model.SelectedItem, model.PathFormat, model.Config.Bookmarks, model.Location, model.BackStack, model.ForwardStack @>)
+                .toFunc(fun (inputMode, (selectStart, selectLen), selected, pathFormat, bookmarks, location, backStack, forwardStack) ->
```

Comment:
When you change ShowHistory to a first-class event type rather than a sub-type of InputMode, I believe you can take this new code out of this already-burdened multi-binding and put it in a new multi-binding just for the history stuff. You'll need a new model property like "IsShowingLocationHistory" to track its state and clear it in `keyPress` if any event other than forward and back is invoked.